### PR TITLE
scale uvicorn servers

### PR DESCRIPTION
**Pull request by aniketmaurya**
<details>
  <summary><b>Before submitting</b></summary>

- [ ] Was this discussed/agreed via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

</details>

## What does this PR do?

Fixes # (issue).

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in GitHub issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: _requirements/test.txt [hunk 1]
```diff
@@ -11,3 +11,4 @@ lightning >2.0.0
 torch >2.0.0
 transformers
 openai>=1.12.0
+numpy==1.26.4
```
Comment (aniketmaurya) [inline _requirements/test.txt:14]: numpy 2.0 was causing issue on windows

Comment (Borda) [reply _requirements/test.txt:14]:
```suggestion
numpy <2.0
```

File: src/litserve/server.py [hunk 2]
```diff
@@ -327,36 +338,39 @@ def inference_worker(
...
     batch_timeout: float,
     stream: bool,
     workers_setup_status: Dict[str, bool] = None,
 ):
     lit_api.setup(device)
     lit_api.device = device
+
+    print(f"Setup complete for worker {worker_id}.")
+
+    # lit_api.worker_id = worker_id
```
Comment (lantiga) [inline src/litserve/server.py:352]:
```suggestion
```

**Review by Borda — APPROVED** [review]

**Comment by bhimrazy** [comment]
This PR on scaling Uvicorn servers looks like a great feature, @aniketmaurya 🎉
I’m looking forward to exploring it and learning more! 🚀

**Pull request merged**